openmicroscopy/infrastructure
# Replacement docker role

This is a new all-in-one docker role, which replaces `docker-storage` and `docker-advanced`.

I've also renamed the `omero-basedeps` role to `basedeps`.

Example basic playbook:
```
- hosts: docker
  roles:
  - role: docker
    # Optional:
    docker_use_upstream_repo: True
```
This will install the upstream docker with the default configuration (uses loopback devices and the default network)

Example advanced playbook:
```
- hosts: docker-advanced
  roles:
  - role: basedeps
  - role: upgrade-distpackages
  - role: docker
    docker_use_upstream_repo: True
    # Custom LVM storage partitions
    docker_use_custom_storage: True
    docker_vgname: VolGroup00
    docker_poolsize: 20G
    docker_metadatasize: 200M
    docker_basefs: xfs
    docker_volumesize: 10G
    docker_lvfilesystem: xfs
    docker_groupmembers: userx
    # Use the standard docker bridge but use a subset of the IPs for testing
    docker_use_custom_network: True
    docker_bridge_name: docker0
    docker_bridge_ips: 172.17.123.0/24
    # Try a local docker DNS setup
  - role: docker-dns-server
    docker_dns:
      forwarders: 8.8.8.8:53,8.8.4.4:53
      domain: internal
      etcd: etcd1
  - role: docker-dns-client
    docker_dns:
      domain: internal
      etcd: etcd1
```
This installs upstream docker, configures the docker partitions (for the docker containers, and also `/var/lib/docker`), modifies the network config slightly so only part of the default IP range is used. It also installs a local docker DNS server.

Testing:
- Check `docker info` matches the configured values
- Check the IP assigned to a container `docker run ...` is within the configured IP range
- This should return the (internal) IPs for running containers `dig \*.docker.internal @test.hostname  SRV`


## Original description PR (for the record):

I've renamed the old `docker` roles to `docker-advanced` (which requires `docker-storage`), and added a default `docker` role (requires storage sizes to be set, use default docker networking). Also prefixed all(?) docker vars.

In theory this should run against the existing servers without making changes, in practice when run in check mode it suggests it may... I'm still investigating.

It should be possible to test the `docker` role now on a VM or the new docker server, see roles/docker/README.md
